File: java/parser/src/main/java/gov/lbl/parser/parser/Condition_attributeVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Condition_attribute;
import gov.lbl.parser.domain.Expression;

public class Condition_attributeVisitor extends modelicaBaseVisitor<Condition_attribute> {
    @Override
    public Condition_attribute visitCondition_attribute(modelicaParser.Condition_attributeContext ctx) {
        ExpressionVisitor expressionVisitor = new ExpressionVisitor();
        Expression expression = ctx.expression() == null ? null :ctx.expression().accept(expressionVisitor);
        
        return new Condition_attribute(expression);
    }
}

File: jsParser/parser/Condition_attributeVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Condition_attribute = require('../domain/Condition_attribute')

const ExpressionVisitor = require('./ExpressionVisitor')

class Condition_attributeVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitCondition_attribute (ctx) {
    let expression = null

    if (ctx.expression()) {
      const expressionVisitor = new ExpressionVisitor.ExpressionVisitor()
      expression = expressionVisitor.visitExpression(ctx.expression())
    }

    return new Condition_attribute.Condition_attribute(expression)
  }
};

Condition_attributeVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitCondition_attribute = this.visitCondition_attribute
exports.Condition_attributeVisitor = Condition_attributeVisitor
